### PR TITLE
Allows to load node embeddings into memory with FP16 format

### PR DESCRIPTION
**Pull request by Elnifio**
This PR gives a new command-line argument `--use_fp16` that, when added, allows the user to load node embeddings into memory with FP16 format. With this argument, the memory-mapped node features will be loaded into memory from disk with FP16 format. This allows users to run the training code on large dataset (IGBH-Large & IGBH-Full, for instance, or IGBH-Medium, which requires >128GB memory) on devices with smaller memory size. 

From our current running result, we see that the accuracy is the same when training with FP16 vs. FP32. This is verified on IGBH-Small, IGBH-Medium, and IGBH-Large. 

We need to note that with this argument, we load all node features into the memory by default, regardless of whether `--in_memory` is true. 

Additionally, this PR also fixes the type of command-line argument `--learning_rate`, so that we can actually control the learning rate from bash script, instead of modifying the actual `.py` file.

### CODE REVIEW

**Review by LiSu — APPROVED** [review]
LGTM. Thanks!
